### PR TITLE
Update Consumer to stop when stopIfNecessary() returns exit code

### PR DESCRIPTION
**Pull request by frans-beech-it**
#408

### CODE REVIEW

File: src/Consumer.php [hunk 1]
```diff
@@ -139,7 +139,13 @@ function (AMQPMessage $message) use ($connection, $options, $connectionName, $qu
             // Finally, we will check to see if we have exceeded our memory limits or if
             // the queue should restart based on other indications. If so, we'll stop
             // this worker and let whatever is "monitoring" it restart the process.
-            $this->stopIfNecessary($options, $lastRestart, $this->gotJob ? true : null);
+            $status = $this->stopIfNecessary(
+                $options, $lastRestart, $startTime, $jobsProcessed, $job
+            );
+
+            if (! is_null($status)) {
+                return $this->stop($status);
```
Comment (MorrisonHotel) [inline src/Consumer.php:147]: A void function must not return a value

Comment (frans-beech-it) [reply src/Consumer.php:147]: ah, return is invalid. See parent class from laravel https://github.com/laravel/framework/blob/bc7d4b7fb1a444c1646d7d47a75185b72e78537d/src/Illuminate/Queue/Worker.php#L117

**Comment by vyuldashev** [comment]
Released under v11.1.2. Thanks!

**Comment by frans-beech-it** [comment]
:+1:

**Pull request merged**